src/components/TypeStyles/TypeStyle.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react'
+import cn from 'classnames'
+import type {ComponentProps} from 'react'
+
+// minor example
+export function TypeStyle(props: ComponentProps<'typestyle'>) {
```

Comment:
Since `typestyle` is not a built-in element, `ComponentProps<'typestyle'>` will fail.

If there is a `OMUIComponent`, one can use `ComponentProps` like this:

```js
type OMUIComponent = React.ComponentProps<typeof OMUIComponent>
```

If we need the props of a OMUIComponent later in out codebase, we can fetch that via ComponentProps.